Fix: svelte-check errors in test files due to missing @types/node

## Problem

Running `npm run check` (svelte-check) reports 11 errors in 4 test files related to Node.js types not being recognized:

```
src/components/search/__tests__/SearchBar.test.ts:3:31
Error: Cannot find module 'module' or its corresponding type declarations.

src/components/search/__tests__/SearchResults.test.ts:3:31
Error: Cannot find module 'module' or its corresponding type declarations.

src/stores/__tests__/pwaStore.test.ts:36:27
Error: Cannot find name 'global'.

src/stores/__tests__/pwaStore.test.ts:76:32
Error: Cannot find name 'Buffer'. Do you need to install type definitions for node?

src/stores/__tests__/websocketStore.test.ts:2:31
Error: Cannot find module 'module' or its corresponding type declarations.
```

## Expected Behavior

`npm run check` should pass without errors.

## Solution

Either:
1. Install `@types/node` as a dev dependency and add it to tsconfig
2. Or exclude test files from svelte-check by updating tsconfig/svelte config
3. Or refactor test files to not rely on Node.js globals (`global`, `Buffer`, `createRequire`)

## Affected Files

- `src/components/search/__tests__/SearchBar.test.ts`
- `src/components/search/__tests__/SearchResults.test.ts`
- `src/stores/__tests__/pwaStore.test.ts`
- `src/stores/__tests__/websocketStore.test.ts`

--- a/frontend/tsconfig.json
+++ b/frontend/tsconfig.json
@@ -21,5 +21,5 @@
     }
   },
   "include": ["src/**/*.ts", "src/**/*.svelte"],
-  "exclude": ["node_modules", "dist", "build"]
+  "exclude": ["node_modules", "dist", "build", "src/**/__tests__/**"]
 }

